test/T61.smart_holder.ref

```diff
@@ -0,0 +1,126 @@
+// File: T61_smart_holder.cpp
+#include <T61.smart_holder.hpp> // A
+#include <T61.smart_holder.hpp> // E1
+#include <T61.smart_holder.hpp> // E2_struct
+#include <T61.smart_holder.hpp> // E3_class
+#include <T61.smart_holder.hpp> // fi
+#include <T61.smart_holder.hpp> // fi_instantiated_by_use_in_function
+#include <sstream> // __str__
+
+#include <functional>
+#include <pybind11/smart_holder.h>
```

Comment:
Right now this test will fail since there is no `smart_holder.h` header in our Pybind11 fork... 🤔 - any thoughts how we can workaround this? Looking at https://github.com/pybind/pybind11/blob/smart_holder/include/pybind11/smart_holder.h it seems that it does include original `pybind11.h` but it is not clear why developers have decided to create separate header...

To get started, I see at least the following ways we can work around this issue:
[a] add smart_holder.h symlink pointing to original file to our current Pybind11 fork, - sounds like a error prone solution with some maintenance overhead 
[b] somehow mark this particular test as "special" so compilation failure is ignored
[c] allow user to specify which include to use in place of `smart_holder.h` and instruct test to override that to use standard header

@kliegeois thoughts on this? I am leaning toward [b] unless some there an elegant solution that I missing...

Reply:
I would prefer the option [b], I think it is the cleanest option.

Reply:
@kliegeois I gave this more thoughts and I think I have better idea: how about if we add one more config directive: `pybind11_include_file` to allow user to customize it and in the same time remove automatic changing `pybind11_include_file` value when one or more `smart_holder` config options was specified? - I think that will give very nice and predictable behavior as well as expand possible code use (for example if for some reason someone need to use different pybind11 header for other reasons). And in the same time this will allow us to fully run the test. Thoughts?